tests/testthat/test-recall_data.R

```diff
@@ -0,0 +1,206 @@
+
+})
+
+test_that("recall_by_make using manufacturer has the correct shape",{
+    result <- recall_by_make('Nissan', manufacturer = TRUE)
+    expect_equal(nrow(result), 25)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+    result <- recall_by_make('Nissan', start_year = 1980)
+    expect_gt(nrow(result), 0)
+    result <- recall_by_make('Nissan', end_year = 1980)
+    expect_gt(nrow(result), 0)
+})
+
+test_that("recall_by_make using partial has the correct shape",{
+    result <- recall_by_make('N', partial = TRUE, manufacturer = FALSE)
+    names <- unique(result$`Make name`)
+    expect_gt(length(names), 1)
+    result <- recall_by_make('Nissan', partial = FALSE, manufacturer = FALSE)
+    names <- unique(result$`Make name`)
+    expect_equal(length(names), 1)
+    result <- recall_by_make('Nissan', partial = FALSE, manufacturer = TRUE)
+    names <- unique(result$`Manufacturer Name`)
+    expect_equal(length(names), 1)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+})
+
+test_that("recall_by_model has the correct shape",{
+    result <- recall_by_model('Civic', limit=25, partial=TRUE)
+    expect_type(result, "list")
+    expect_equal(nrow(result), 25)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_col_names <- c("Recall number","Manufacturer Name",
+                            "Model name", "Make name", "Year","Recall date")
+    expect_equal(colnames(result), expected_col_names)
+    expected_dtypes <- c("character", "character", "character",
+                         "character", "integer",   "Date")
+    returned_dtypes <- as.vector(unlist(sapply(result, class)))
+    expect_equal(returned_dtypes, expected_dtypes)
+    result <- recall_by_model('Civic', start_year = 1980)
+    expect_gt(nrow(result), 0)
+    result <- recall_by_model('Civic', end_year = 1980)
+    expect_gt(nrow(result), 0)
+})
+
+test_that("recall_by_model by year range has the correct shape",{
+    result <- recall_by_model('Civic', start_year = 1980, end_year=2005)
+    expect_gt(nrow(result), 0)
```

Comment:
probably just wrap into the above as you're already testing start_year and end_year